cmd/pulumi-resource-docker-buildkit/main.go

```diff
@@ -333,62 +369,24 @@ func (w *logWriter) Write(p []byte) (n int, err error) {
 }
 
 func hashContext(contextPath string) (string, error) {
+	context, err := buildContext(contextPath)
```

Comment:
You get this buildContext multiple times. Couldn't it change between attempts? That could cause your hash to not match the actual context used when building the image.

Reply:
I mean, ideally it shouldn't - that's the content-addressable storage future we were promised. At any rate, I think it _ought_ to be safe, provided docker's archives are stable. A big bet, if unvalidated, but I think it's directionally right. We can still redo the hash computation from earlier (but that has a problem with renamed files) (: